bin.src/imsim.py

```diff
@@ -89,26 +93,22 @@ def main():
                 detector_list.append(make_galsim_detector(camera_wrapper, det.getName(),
                                                           phot_params, obs_md))
 
-    # Add noise and sky background
-    # The simple code using the default lsst-GalSim interface would be:
-    #
-    #    PhoSimStarCatalog.noise_and_background = ExampleCCDNoise(addNoise=True,
-    #                                                             addBackground=True)
-    #
-    # But, we need a more realistic sky model and we need to pass more than
-    # this basic info to use Peter Y's ESO sky model.
-    # We must pass obs_metadata, chip information etc...
+    apply_sensor_model = not arguments.disable_sensor_model
     noise_and_background \
-        = ESOSkyModel(obs_md, addNoise=True, addBackground=True)
+        = make_sky_model(obs_md, phot_params, addNoise=True, addBackground=True,
+                         apply_sensor_model=apply_sensor_model,
+                         logger=logger)
 
     bp_dict = BandpassDict.loadTotalBandpassesFromFiles(bandpassNames=obs_md.bandpass)
 
-    gs_interpreter = GalSimInterpreter(obs_metadata=obs_md,
-                                       epoch=2000.0,
-                                       detectors=detector_list,
-                                       bandpassDict=bp_dict,
-                                       noiseWrapper=noise_and_background,
-                                       seed=arguments.seed)
+    gs_interpreter = make_gs_interpreter(obs_md, detector_list, bp_dict,
+                                         noise_and_background,
+                                         epoch=2000.0,
+                                         seed=arguments.seed,
+                                         apply_sensor_model=apply_sensor_model)
+    chip_name = arguments.sensor if arguments.sensor is not None\
+                else "R:2,2 S:1,1"
```

Comment:
So this says if we don't specify a sensor then we select the middle sensor in the focal plane.  So if someone gives imSim a instance catalog without specifying the sensors and the sources in the catalog fall on another sensor the output will be this sensor with only the sky correct?  

Is this consistent with the parameter help for --sensor: "'If None, then simulate all sensors with sources on them'"

Reply:
Setting `chip_name` here is only used in the next line where the `.sky_bg_per_pixel` attribute is set, and that attribute is only used to set the scale used for the `folding_threshold` that the `galsim` code uses to compute the postage stamp size for point sources.  In that application, a precise value really isn't needed, so the center of the focalplane is good enough.  In all other cases, including when simulating all of the sensors, the code in `skyModel.py` uses the sensor-specific value.  The problem is that we need to set this attribute before calling `.drawObject` below.